generator/osm_source.cpp

```diff
@@ -524,6 +532,30 @@ class MainFeaturesEmitter : public EmitterBase
 };
 }  // anonymous namespace
 
+void SyncOfstream::Open(string const & fullPath)
+{
+  lock_guard<mutex> gard(m_mutex);
+  m_stream.open(fullPath, std::ofstream::out);
+}
+
+bool SyncOfstream::IsOpened()
+{
+  lock_guard<mutex> gard(m_mutex);
+  return m_stream.is_open() && !m_stream.fail();
+}
+
+void SyncOfstream::Write(uint32_t featureId, vector<osm::Id> const & osmIds)
+{
+  if (!IsOpened())
+    return;
+
+  lock_guard<mutex> gard(m_mutex);
+  m_stream << featureId << ", ";
```

Comment:
At least at development stage I'd prefer to leave it as ASCII text files. They are not very large (up to 10K lines) and it's convenient to have them as text for debugging restrictions. Then we can change serializing/deserializing to binary encoding.